pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/MessageIdImpl.java

```diff
@@ -77,8 +77,7 @@ public boolean equals(Object obj) {
 
     @Override
     public String toString() {
-        return Objects.toStringHelper(MessageIdImpl.class).add("ledgerId", ledgerId).add("entryId", entryId)
-                .add("partitionIndex", partitionIndex).toString();
+        return String.format("%d:%d:%d", ledgerId, entryId, partitionIndex);
```

Comment:
I and @rdhabalia discussed this - if I suppress the partition index and I see {1:2:3} in the log it's difficult to figure out if 3 is the partitionIndex or batch Message index.
To avoid this confusion I can print the name of the class in toString method, let me know if you are ok with putting the class name in toString method